### ISSUE TITLE
Erlang error %Phoenix.Router.NoRouteError reported on 404 after upgrade to plug_cowboy 2

### ISSUE THREAD

**Issue by silviurosu** [jarednorman/bugsnag-elixir#85]
I am copying this bug report from bugsnex library since the issue is the same. It was reported by @adamniedzielski: 

After upgrade to plug_cowboy 2, 404 responses started to appear as errors in Bugsnag:
```
Erlang error: {{%Phoenix.Router.NoRouteError{conn: %Plug.Conn{adapter: {Plug.Cowboy.Conn, :...}, assigns: %{}, before_send: [#Function<1.1812729/1 in Plug.Logger.call/2>], body_params: %{}, cookies: %Plug.Conn.Unfetched{aspect: :cookies}, halted: false, host: "fulfillment-staging.liefery.com", method: "GET", owner: #PID<0.2309.0>, params: %{}, path_info: ["css", "vendor.css"], path_params: %{}, port: 80, private: %{Fulfillment.Router => {[], %{}}, :phoenix_endpoint => Fulfillment.Endpoint, :phoenix_router => Fulfillment.Router, :plug_session_fetch => #Function<1.58261320/1 in Plug.Session.fetch_session/1>}, query_params: %{}, query_string: "", remote_ip: {10, 10, 1, 5}, req_cookies: %Plug.Conn.Unfetched{aspect: :cookies}, req_headers: [{"accept", "text/css,*/*;q=0.1"}, {"accept-encoding", "gzip, deflate, br"}, {"accept-language", "en-US,en;q=0.9,de;q=0.8"}
```

This didn't happen when using plug_cowboy 1.

It looks like multiple other libraries for error tracking services experienced the similar problem:

[appsignal/appsignal-elixir#417](https://github.com/appsignal/appsignal-elixir/issues/417)
[getsentry/sentry-elixir#328](https://github.com/getsentry/sentry-elixir/issues/328)
[honeybadger-io/honeybadger-elixir#193](https://github.com/honeybadger-io/honeybadger-elixir/issues/193)
I found the content in [appsignal/appsignal-elixir#417](https://github.com/appsignal/appsignal-elixir/issues/417) particularly informative. [Here](https://github.com/appsignal/appsignal-elixir/issues/417#issuecomment-439876201) is the explanation and [here](https://github.com/appsignal/appsignal-elixir/issues/417#issuecomment-443695327) is a suggestion to switch from listening to Erlang's error_logger to listening to Elixir's Logger.

**Comment by hrlmartins** [comment]
Is it possible to filter this error out using the exception_filter config with a custom module? Thanks!

**Comment by smaximov** [comment]
@hrlmartins, yes, the example in the [`exception_filter` section](https://github.com/jarednorman/bugsnag-elixir/tree/ebf3569ada87e27f83bc21935fb96f4aab82b23d#exception_filter) in README should do.

**Comment by grzuy** [comment]
> I found the content in [appsignal/appsignal-elixir#417](https://github.com/appsignal/appsignal-elixir/issues/417) particularly informative. [Here](https://github.com/appsignal/appsignal-elixir/issues/417#issuecomment-439876201) is the explanation and [here](https://github.com/appsignal/appsignal-elixir/issues/417#issuecomment-443695327) is a suggestion to switch from listening to Erlang's error_logger to listening to Elixir's Logger.

Indeed, and that is discussed and reported it https://github.com/bugsnag-elixir/bugsnag-elixir/issues/120.
